### PR TITLE
[SPARK-33479][DOC][FOLLOWUP] DocSearch: Support filtering search results by version 

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

In the discussion https://github.com/apache/spark/pull/30292#issuecomment-725613417, we planned to apply a new API key for each Spark release. However, it turns that DocSearch supports crawling multiple URLs from one website and filtering by fact key: https://docsearch.algolia.com/docs/config-file/#using-regular-expressions

Thanks to the help from @shortcuts, our Spark doc supports multiple version now: https://github.com/algolia/docsearch-configs/pull/2868

This PR is to add the fact key in the search script and update the instruction in the comment.

### Why are the changes needed?

To support filtering Spark documentation search results by the current document version. 

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Manual test

### CODE REVIEW

**Comment by gengliangwang** [comment]
cc @shortcuts @maropu @sarutak @HyukjinKwon @beliefer

File: docs/_config.yml [hunk 1]
```diff
@@ -26,15 +26,20 @@ SCALA_VERSION: "2.12.10"
...
 SPARK_ISSUE_TRACKER_URL: https://issues.apache.org/jira/browse/SPARK
 SPARK_GITHUB_URL: https://github.com/apache/spark
-# Before a new release, we should apply a new `apiKey` for the new Spark documentation
-# on https://docsearch.algolia.com/. Otherwise, after release, the search results are always based
-# on the latest documentation(https://spark.apache.org/docs/latest/) even when visiting the
-# documentation of previous releases.
+# Before a new release, we should:
+#   1. update the `version` array for the new Spark documentation
+#      on https://github.com/algolia/docsearch-configs/blob/master/configs/apache_spark.json.
+#   2. update the value of `facetFilters.version` in `algoliaOptions` on the new release branch.
```
Comment (gengliangwang) [inline docs/_config.yml:32]: An alternative way is always to update the current spark doc. For example, if we are going to release 3.1.1 and the current doc version is 3.1.0, we can update the version of http://spark.apache.org/docs/3.1.0 from `latest` to `3.1.0`. For 3.1.1, we can still use the version `latest` until the next relealse.

**Comment by shortcuts** [comment]
Hi guys,

Feel free to submit a PR whenever you need to add a version (as it only supports `latest` for now), I'll approve it.

**Review by HyukjinKwon — APPROVED** [review]

**Review by maropu — APPROVED** [review]

**Comment by maropu** [comment]
Btw, any plan to provide this search functionality for doc v2.4/v3.0, too? This is very useful for users, I think.

**Comment by maropu** [comment]
Thanks! Merged to master.

**Comment by HyukjinKwon** [comment]
I'm good with porting back. Its a doc only change in the end

**Pull request closed without merging**